src/main/java/com/example/ludogoriesoft/lukeriaerpapi/services/InvoiceService.java

```diff
@@ -18,6 +19,25 @@ public class InvoiceService {
     private final InvoiceRepository invoiceRepository;
     private final ModelMapper modelMapper;
 
+    public Long findLastInvoiceNumberStartingWithTwo()  {
+        String prefix = "2";
+
+        List<String> lastInvoiceNumbers = invoiceRepository.findLastInvoiceNumberStartingWith(prefix);
+
+        if (lastInvoiceNumbers.isEmpty()) {
+           return 2000000000L;
+        }
+
+        String maxLastDigitNumber = lastInvoiceNumbers.stream()
+                .max(Comparator.comparing(this::getLastDigit))
+                .orElse(null);
+
+        return Long.parseLong(maxLastDigitNumber);
```

Comment:
If maxLastDigitNumber is null, an exception will be thrown because it can't be parsed

Reply:
maybe we can use something like "orElse("0")